pkg/service/well-known/did_configuration.go

```diff
@@ -33,6 +56,80 @@ func (s DIDConfigurationService) Status() svcframework.Status {
 
 var _ svcframework.Service = (*DIDConfigurationService)(nil)
 
+func (s DIDConfigurationService) VerifyDIDConfiguration(ctx context.Context, req *VerifyDIDConfigurationRequest) (*VerifyDIDConfigurationResponse, error) {
+	location := req.Origin + DIDConfigurationLocationSuffix
+	httpReq, err := http.NewRequestWithContext(ctx, http.MethodGet, location, nil)
+	if err != nil {
+		return nil, errors.Wrap(err, "creating http request")
+	}
+	if httpReq.URL.Scheme != "https" {
+		return nil, errors.Errorf("origin expected to be https but got %s", req.Origin)
+	}
+
+	httpResponse, err := s.HTTPClient.Do(httpReq)
+	defer func(Body io.ReadCloser) {
+		_ = Body.Close()
+	}(httpResponse.Body)
+	if err != nil {
+		return nil, errors.Wrap(err, "performing http request")
+	}
+
+	if !util.Is2xxResponse(httpResponse.StatusCode) {
+		return nil, errors.Errorf("expected 2xx code, got %d", httpResponse.StatusCode)
+	}
+
+	bodyCopy := new(bytes.Buffer)
+	bodyTeeReader := io.TeeReader(httpResponse.Body, bodyCopy)
+
+	var didConfiguration DIDConfiguration
+	if err := json.NewDecoder(bodyTeeReader).Decode(&didConfiguration); err != nil {
+		return nil, errors.Wrap(err, "decoding body")
+	}
+
+	response := VerifyDIDConfigurationResponse{
+		DIDConfiguration: bodyCopy.String(),
+	}
+	for _, domainLinkageCredential := range didConfiguration.LinkedDIDs {
+		// For a Domain Linkage Credential to be deemed valid, it MUST be successfully processed in accordance with the following steps:
+		//
+		// 1. The credentialSubject.id MUST be a DID, and the value MUST be equal to both the Subject and Issuer of the Domain Linkage Credential.
+		credentialSubjectID := domainLinkageCredential.Credential.CredentialSubject["id"].(string)
+		if !strings.HasPrefix(credentialSubjectID, "did:") {
```

Comment:
believe there's a method `IsValidDID` or similar in the utils somewhere

Reply:
I searched for something similar, but couldn't find anything.